webpmux.cpp

```diff
@@ -0,0 +1,90 @@
+#include "webpmux.hpp"
+#include <webp/encode.h>
+
+extern "C" {
+
+WebPAnimDecoder* webpmux_create_decoder(void* buf, size_t buf_len, void* info)
+{
+    auto i = static_cast<WebPAnimInfo*>(info);
+    WebPData data = {static_cast<const uint8_t*>(buf), buf_len};
+    WebPAnimDecoderOptions dec_options;
+    WebPAnimDecoderOptionsInit(&dec_options);
+    dec_options.color_mode = MODE_BGRA;
+    WebPAnimDecoder* dec = WebPAnimDecoderNew(&data, &dec_options);
+    if (dec == NULL)
+        return NULL;
+    WebPAnimDecoderGetInfo(dec, i);
+    return dec;
+}
+
+WebPAnimEncoder* webpmux_create_encoder(int width, int height)
+{
+    WebPAnimEncoderOptions enc_options;
+    WebPAnimEncoderOptionsInit(&enc_options);
+    return WebPAnimEncoderNew(width, height, &enc_options);
+}
+
+int webpmux_decoder_read_data(WebPAnimDecoder* dec, opencv_mat mat)
+{
+    auto m = static_cast<cv::Mat*>(mat);
+    int timestamp = 0;
+    uint8_t* buf = NULL;
+    int res = WebPAnimDecoderGetNext(dec, &buf, &timestamp);
+    if (!res)
+        return -1;
+    cv::Mat img(cv::Size(m->cols, m->rows), CV_8UC4, buf);
+    img.copyTo(*m);
```

Comment:
looks like opencv uses `WebPDecodeBGRInto` to avoid extra alloc and copy here. i looked at animated webp code and it doesn't seem to really support that :( probably our best option would be to copy a lot of `src/demux/anim_decode.c` into here and reuse the buffers we have to avoid the allocations/copies